File: 2324-find-triangular-sum-of-an-array/2324-find-triangular-sum-of-an-array.cpp
class Solution {
public:
     
    int triangularSum(vector<int>& nums) {
        while(nums.size() > 1){
            vector<int> temp;
        
            for(int j = 0; j < nums.size() -1; j++){
                temp.push_back((nums[j]+nums[j+1]) %10) ;
            }
            nums= move(temp); 
        }
        return nums[0];
    }
};

File: 2324-find-triangular-sum-of-an-array/2324-find-triangular-sum-of-an-array.java
class Solution {
    public int triangularSum(int[] nums) {
        // for( int i = nums.length - 1 ; i > 0; i--){
        //     for(int j = 0; j < i ; j++){
        //         nums[j] = (nums[j] + nums[j+1]) % 10;
        //     }
        // }
        // return nums[0];
        int n = nums.length;

        while (n > 1) {  
            for (int i = 0; i < n - 1; i++) {
                nums[i] = (nums[i] + nums[i + 1]) % 10;
            }
            n--; 
        }

        return nums[0];
    }
}